angular-swfobject.js

```diff
@@ -46,23 +46,37 @@ angular.module('swfobject', [])
           });
         }
 
-
         // http://learnswfobject.com/advanced-topics/executing-javascript-when-the-swf-has-finished-loading/
         function swfLoadEvent(evt, fn) {
-            //Set up a timer to periodically check value of PercentLoaded
-            var loadCheckInterval = $interval(function () {
-              //Ensure Flash Player's PercentLoaded method is available and returns a value
-              if (typeof evt.ref.PercentLoaded !== "undefined" && evt.ref.PercentLoaded()) {
-                //Once value == 100 (fully loaded) we can do whatever we want
-                if (evt.ref.PercentLoaded() === 100) {
-                  //Clear interval
-                  $interval.cancel(loadCheckInterval);
-                  loadCheckInterval = null;
-                  //Execute function
-                  fn({evt: evt});
+            //This interval ensures we don't try to access PercentLoaded too soon
+            var startCheckInterval = $interval(function () {
+                //Ensure Flash Player's PercentLoaded method is available
+                if (typeof evt.ref.PercentLoaded !== 'undefined') {
+                    $interval.cancel(startCheckInterval);
+                    startCheckInterval = null;
```

Comment:
We could probably avoid all this interval checking and cancelling by using timeouts instead?